### PR TITLE
Add METR-LA dataset

### PR DESCRIPTION
**Pull request by aurorarossi**
This PR adds the dataset used in [this paper](https://arxiv.org/pdf/1707.01926.pdf).

### CODE REVIEW

File: src/MLDatasets.jl [hunk 1]
```diff
@@ -12,6 +12,8 @@ using DelimitedFiles: readdlm
 using FileIO
 import CSV
 using LazyModules: @lazy
+using NPZ: npzread
```
Comment (Dsantra92) [inline src/MLDatasets.jl:15]: Does NPZ need to a be hard import?

Comment (aurorarossi) [reply src/MLDatasets.jl:15]: No, we can use `@ require` both for NPZ and Statistics.

**Comment by CarloLucibello** [comment]
The CI failure is unrelated. Can you change the UD_English checksum to 2e94de3333b3b17bd06769ff8c4a81896fabec3ae51a6366f7af82625679f561?

File: src/MLDatasets.jl [hunk 2]
```diff
@@ -25,8 +25,10 @@ include("require.jl") # export @require
 @require import DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
 @require import ImageShow = "4e3cecfd-b093-5904-9786-8bbb286a6a31"
 @require import Chemfiles = "46823bd8-5fb3-5f92-9aa0-96921f3dd015"
+@require import NPZ = "15e1cf62-19b3-5cfa-8e77-841668bca605"
+@require import Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
```
Comment (CarloLucibello) [inline src/MLDatasets.jl:29]: we can just normally import Statistics, it is a standard library with little overhead

File: src/datasets/graphs/metrla.jl [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    return dir
+end
+
+function read_metrla(d::String)
+    adj_matrix = NPZ.npzread(joinpath(d, "adj_mat.npy"))
+    node_features = NPZ.npzread(joinpath(d, "node_values.npy"))
+    return adj_matrix, node_features
+end
+    
+function generate_task(node_values::AbstractArray, num_timesteps_in::Int, num_timesteps_out::Int)
```
Comment (CarloLucibello) [inline src/datasets/graphs/metrla.jl:72]:
```suggestion
function metrla_generate_task(node_values::AbstractArray, num_timesteps_in::Int, num_timesteps_out::Int)
```

File: src/datasets/graphs/metrla.jl [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    indices = [(i, i + num_timesteps_in + num_timesteps_out) for i in 1:(size(node_values,1) - num_timesteps_in - num_timesteps_out)]
+    features=[]
```
Comment (CarloLucibello) [inline src/datasets/graphs/metrla.jl:74]:
```suggestion
    features = []
```

File: src/datasets/graphs/metrla.jl [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    targets=[]
+    for (i,j) in indices
+        push!(features,node_values[i:i+num_timesteps_in-1,:,:])
```
Comment (CarloLucibello) [inline src/datasets/graphs/metrla.jl:77]:
```suggestion
        push!(features, node_values[i:i+num_timesteps_in-1,:,:])
```

File: src/graph.jl [hunk 6]
```diff
@@ -226,3 +226,30 @@ function edgeindex2adjlist(s, t, num_nodes; inneigs = false)
     end
     return adj
 end
+
+function adjmatrix2edgeindex(adj; weighted = true, inneigs = false)
+    s, t = Int[], Int[]
+    if weighted 
+        w = Float64[]
+    end
```
Comment (CarloLucibello) [inline src/graph.jl:234]:
```suggestion
function adjmatrix2edgeindex(adj::AbstractMatrix{T}; weighted = true, inneigs = false) where T
    s, t = Int[], Int[]
    if weighted 
        w = T[]
    end
```

**Pull request merged**